elixir/series/lib/string_series.ex

```diff
@@ -13,9 +13,22 @@ defmodule StringSeries do
     cond do
       String.length(string) == n ->
         [string]
+
       String.length(string) < n ->
         []
+
+      String.length(string) > n ->
+        count = String.length(string) - n + 1
+        process_string(string, n, count)
     end
   end
 
+  def process_string(_string, n, 0), do: []
+
+  def process_string(string, n, count) do
+    [
+      String.slice(string, 0, n)
+      | process_string(String.slice(string, 1, String.length(string) - 1), n, count - 1)
+    ]
+  end
```

Comment:
```suggestion
  def process_string(string, n, count) do
    new_length = String.length(string) - 1

    process_substring =
      string
      |> String.slice(1, new_length)
      |> process_string(n, count - 1)

    [String.slice(string, 0, n) | process_substring]
  end
```

Line 31 is getting a little hairy. Some variables could make it more easily parseable

Reply:
Thanks for this suggestion. I personally find the process_substring solution harder to read and understand.
But I implemented your idea with new_length